tx-parser/benches/bench.rs

```diff
@@ -0,0 +1,55 @@
+#![feature(test)]
+
+// These benchmarks can be run from the project root with:
+// > cargo bench --package mentat_tx_parser
+
+extern crate test;
+extern crate edn;
+extern crate mentat_tx_parser;
+
+use test::Bencher;
+use mentat_tx_parser::Tx;
+
+#[bench]
+fn bench_parse1(b: &mut Bencher) {
+    let input = r#"[[:db/add 1 :test/val "a"]]"#;
+    let parsed_edn = edn::parse::value(input).expect("to parse test input");
+    b.iter(|| {
+        return Tx::parse(parsed_edn.clone());
```

Comment:
> Concern: we're cloning the string here, inside the benchmark. Is it fairer to clone it outside?

That seems ideal, but without the clone we get this error:
```
53 |         return Tx::parse(parsed_edn);
   |                          ^^^^^^^^^^ cannot move out of captured outer variable in an `FnMut` closure
```

> Or is it more realistic to call edn::parse inside the loop, and not explicitly clone?

We could - I thought cloning the parsed edn would be more narrow than doing the actual edn parsing in the loop.  If we wanted to fold edn parsing into this benchmark that'd be OK but it turns into a broader test - I figured we might want a separate benchmark for just parsing out the EDN.

With my test script and edn parsing in the loop I get:

```
bench_parse1      6,814 ns/iter (+/- 2,250)
bench_parse2    604,382 ns/iter (+/- 70,221)
---
bench_parse1      7,526 ns/iter (+/- 2,517)
bench_parse2    631,710 ns/iter (+/- 86,643)
---
bench_parse1      7,073 ns/iter (+/- 2,645)
bench_parse2    624,657 ns/iter (+/- 74,293)
---
bench_parse1      7,124 ns/iter (+/- 2,509)
bench_parse2    636,105 ns/iter (+/- 83,272)
---
bench_parse1      7,118 ns/iter (+/- 3,830)
bench_parse2    620,871 ns/iter (+/- 116,928)
```

And with the clone in the loop and parsing outside I get:

```
---
bench_parse1      6,925 ns/iter (+/- 1,087)
bench_parse2    382,473 ns/iter (+/- 63,815)
---
bench_parse1      7,073 ns/iter (+/- 1,339)
bench_parse2    428,671 ns/iter (+/- 185,621)
---
bench_parse1      7,392 ns/iter (+/- 2,354)
bench_parse2    398,395 ns/iter (+/- 134,769)
---
bench_parse1      7,044 ns/iter (+/- 792)
bench_parse2    394,561 ns/iter (+/- 44,045)
---
bench_parse1      6,685 ns/iter (+/- 1,071)
bench_parse2    387,890 ns/iter (+/- 70,388)
```